caravel/assets/javascripts/SqlLab/components/QueryAutoRefresh.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import moment from 'moment';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import * as Actions from '../actions';
+
+$ = require('jquery');
+
+
+class QueryAutoRefresh extends React.Component {
+  componentWillMount() {
+    this.startTimer();
+  }
+  componentWillUnmount() {
+    this.stopTimer();
+  }
+  startTimer() {
+    if (!(this.timer)) {
+      this.timer = setInterval(this.stopwatch.bind(this), 2000);
+    }
+  }
+  stopTimer() {
+    clearInterval(this.timer);
+    this.timer = null;
+  }
+  stopwatch() {
+    const url = '/caravel/queries/0';
+    $.getJSON(
```

Comment:
we should maintain a `isConnected` in the state from here (and the error counterpart) so that we can surface it on the UI when the user go offline